dist/angular-promise-buttons.js

```diff
@@ -23,47 +23,110 @@ angular.module('angularPromiseButtons')
+                    }, waitTime);
+                }
+
+                function setFinishedState(btnEl, onEndConfig)
+                {
+                    if (onEndConfig)
+                    {
+                        if (onEndConfig.resultHtml)
+                        {
+                            btnEl.html(onEndConfig.resultHtml);
+                        }
+                        if (onEndConfig.resultCssClass)
+                        {
+                            btnEl.addClass(onEndConfig.resultCssClass);
+                        }
+                    }
                 }
 
-                function handleLoadingFinished(btnEl)
+                function removeLoadingState(btnEl)
                 {
                     if (cfg.btnLoadingClass) {
                         btnEl.removeClass(cfg.btnLoadingClass);
                     }
+                }
+
+                function revertToNormalState(btnEl, defaultHtml, onEndConfig)
+                {
                     if (cfg.disableBtn) {
                         btnEl.removeAttr('disabled');
                     }
+                    if (defaultHtml) {
+                        btnEl.html(defaultHtml);
+                    }
+                    if (onEndConfig && onEndConfig.resultCssClass)
+                    {
+                        btnEl.removeClass(onEndConfig.resultCssClass);
+                    }
                 }
 
                 function initPromiseWatcher(watchExpressionForPromise, btnEl)
                 {
                     // watch promise to resolve or fail
                     scope.$watch(watchExpressionForPromise, function (mVal)
                     {
+                        var initPromise = null;
                         // for regular promises
-                        if (mVal && mVal.then) {
-                            handleLoading(btnEl);
-                            mVal.finally(function ()
-                            {
-                                handleLoadingFinished(btnEl);
-                            });
+                        if (mVal && mVal.then)
+                        {
+                            initPromise = mVal;
                         }
                         // for $resource
-                        else if (mVal && mVal.$promise) {
-                            handleLoading(btnEl);
-                            mVal.$promise.finally(function ()
-                            {
-                                handleLoadingFinished(btnEl);
-                            });
+                        else if (mVal && mVal.$promise)
+                        {
+                            initPromise = mVal.$promise;
+                        }
+
+                        if (initPromise)
+                        {
+                            var defaultHtml = cfg.defaultHtml || btnEl.html();
+                            setLoadingState(btnEl);
+                            initPromise.then(
```

Comment:
Have you checked if the `then` and `catch` callbacks are still working for the promise, when you use it in a controller?